compiler/modules/front/const.bal

```diff
@@ -26,11 +26,14 @@ function resolveConstDefn(ModuleSymbols mod, s:ConstDefn defn) returns s:Resolve
     }
 }
 
-function resolveConstExpr(ModuleSymbols mod, s:ModuleLevelDefn defn, s:Expr expr, t:SemType? expectedType) returns s:ResolvedConst|ResolveTypeError {
+function resolveConstExpr(ModuleSymbols mod, s:ModuleLevelDefn defn, s:Expr expr, t:SemType? expectedType, string? td = ()) returns s:ResolvedConst|ResolveTypeError {
     ExprContext cx = new ExprContext(mod, defn, constCode, constEnvironment, ());
     var { result } = check codeGenExpr(cx, constBasicBlock, expectedType, expr);
     bir:ConstOperand operand = <bir:ConstOperand>result;
     if expectedType != () && !t:isSubtype(mod.tc, operand.semType, expectedType) {
+        if td != () && expr is s:VarRefExpr {
+            return err:semantic(`initializer ${expr.name} is not ${td}`, s:locationInDefn(defn, expr.startPos));
```

Comment:
This still doesn't seem good to me: 

Why is the test for VarRefExpr right?  I guess you are using that so that you can expr.name. But that's not really the right thing to test.

We are calling resolveConstExpr in the following cases

1. RHS of const definition
2. In a match pattern in a match stmt
3. In a type desc for a singleton type
4. In an array length in an array type desc

The current error message is really only right for case (1).

What has VarRefExpr got to do with any of these?

Please propose a better design. We can discuss in today's call.